packages/flutter_tools/lib/src/web/devfs_config.dart

```diff
@@ -0,0 +1,261 @@
+      );
+    }
+    return cliValue;
+  }
+  return fileValue;
+}
+
+Future<WebDevServerConfig> loadDevConfig({
+  String? hostname,
+  String? port,
+  String? tlsCertPath,
+  String? tlsCertKeyPath,
+  Map<String, String>? headers,
+  int? debugPort,
+  List<String>? browserFlags,
+}) async {
+  final fs.File devConfigFile = globals.fs.file(devConfigFilePath);
+  var fileConfig = const WebDevServerConfig();
+
+  if (!devConfigFile.existsSync()) {
+    globals.printStatus('No $devConfigFilePath found');
+  } else {
+    try {
+      final String devConfigContent = await devConfigFile.readAsString();
+      final YamlDocument yamlDoc = loadYamlDocument(devConfigContent);
+      final YamlNode contents = yamlDoc.contents;
+      if (contents is! YamlMap) {
+        throw YamlException(
+          '$devConfigFilePath file found, but it must be a YAML map (e.g., "server:"). Found a ${contents.runtimeType} instead.',
+          contents.span,
+        );
+      }
+
+      if (!contents.containsKey('server') || contents['server'] is! YamlMap) {
+        // final SourceSpan span = (contents.containsKey('server') && contents['server'] is YamlNode)
+        //     ? (contents['server'] as YamlNode).span
+        //     : contents.span;
+        throw YamlException(
+          '"$devConfigFilePath" file found, but the "server" key is missing or malformed. It must be a YAML map.',
+          null,
+        );
+      }
+
+      final serverYaml = contents['server'] as YamlMap;
+      fileConfig = WebDevServerConfig.fromYaml(serverYaml);
+      globals.printStatus('\nLoaded configuration from $devConfigFilePath');
+      globals.printTrace(fileConfig.toString());
+    } on YamlException catch (e) {
+      globals.printError('Error: Failed to parse $devConfigFilePath: ${e.message} ${e.span}');
+      rethrow;
+    } on Exception catch (e) {
+      globals.printError('An unexpected error occurred while reading $devConfigFilePath: $e');
+      globals.printStatus(
+        'Reverting to default flutter_tools web server configuration due to unexpected error.',
+      );
+    }
+  }
+
+  final String finalHost =
+      _getOverriddenValue<String>('host', fileConfig.host, hostname) ?? 'localhost';
+  final int? finalPort = _getOverriddenValue<int>(
+    'port',
+    fileConfig.port,
+    port != null ? int.tryParse(port) : null,
+  );
+  final String? finalCertPath = _getOverriddenValue<String>(
+    'TLS cert path',
+    fileConfig.https?.certPath,
+    tlsCertPath,
+  );
+  final String? finalCertKeyPath = _getOverriddenValue<String>(
+    'TLS cert key path',
+    fileConfig.https?.certKeyPath,
+    tlsCertKeyPath,
+  );
+  HttpsConfig? finalHttpsConfig;
+  if (finalCertPath != null || finalCertKeyPath != null || fileConfig.https != null) {
+    finalHttpsConfig = HttpsConfig(certPath: finalCertPath, certKeyPath: finalCertKeyPath);
+  } else {
+    finalHttpsConfig = null;
+  }
```

Comment:
This looks like a private `HttpsConfig? _loadHttpsConfig` method, so we can split `loadWebDevServerConfig` into smaller chunks.